gstools/tools/export.py

```diff
@@ -90,9 +120,75 @@ def vtk_export_unstructured(filename, pos, fields):  # pragma: no cover
         ):
             raise ValueError(
                 "gstools.vtk_export_unstructured: "
-                + "field shape doesn't match the given mesh"
+                "field shape doesn't match the given mesh"
             )
-    pointsToVTK(filename, x, y, z, data=fields)
+    return x, y, z, fields
+
+
+def to_vtk_unstructured(pos, fields):  # pragma: no cover
+    """Export a field to vtk structured rectilinear grid file.
+
+    Parameters
+    ----------
+    pos : :class:`list`
+        the position tuple, containing main direction and transversal
+        directions
+    fields : :class:`dict` or :class:`numpy.ndarray`
+        Unstructured fields to be saved.
+        Either a single numpy array as returned by SRF,
+        or a dictionary of fields with theirs names as keys.
+    """
+    x, y, z, fields = _vtk_unstructured_helper(pos=pos, fields=fields)
+    try:
+        import pyvista as pv
+        grid = pv.PolyData(np.c_[x, y, z]).cast_to_unstructured_grid()
+        grid.point_arrays.update(fields)
+    except ImportError:
+        raise ImportError("Please install PyVista to create VTK datasets.")
+    return grid
+
+
+def vtk_export_unstructured(filename, pos, fields):  # pragma: no cover
+    """Export a field to vtk unstructured grid file.
+
+    Parameters
+    ----------
+    filename : :class:`str`
+        Filename of the file to be saved, including the path. Note that an
+        ending (.vtu) will be added to the name.
+    pos : :class:`list`
+        the position tuple, containing main direction and transversal
+        directions
+    fields : :class:`dict` or :class:`numpy.ndarray`
+        Unstructured fields to be saved.
+        Either a single numpy array as returned by SRF,
+        or a dictionary of fields with theirs names as keys.
```

Comment:
Please provide a description of what is being returned.

Reply:
Same no return type. Simply good practice